Userland/Applications/PixelPaint/HistogramWidget.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2022, Torsten Engelmann
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "HistogramWidget.h"
+#include "Image.h"
+#include "ImageEditor.h"
+#include "Layer.h"
+#include <LibGUI/Painter.h>
+#include <LibGfx/Palette.h>
+#include <LibGfx/Path.h>
+
+REGISTER_WIDGET(PixelPaint, HistogramWidget);
+
+namespace PixelPaint {
+
+HistogramWidget::HistogramWidget()
+{
+    set_height(65);
+}
+
+HistogramWidget::~HistogramWidget()
+{
+    if (m_image)
+        m_image->remove_client(*this);
+}
+
+void HistogramWidget::set_image(Image* image)
+{
+    if (m_image == image)
+        return;
+    if (m_image)
+        m_image->remove_client(*this);
+    m_image = image;
+    if (m_image)
+        m_image->add_client(*this);
+
+    (void)rebuild_histogram_data();
+}
+
+ErrorOr<void> HistogramWidget::rebuild_histogram_data()
+{
+    if (m_image) {
+        auto full_bitmap = TRY(m_image->try_compose_bitmap(Gfx::BitmapFormat::BGRA8888));
+
+        m_data.red.clear_with_capacity();
+        m_data.green.clear_with_capacity();
+        m_data.blue.clear_with_capacity();
+        m_data.brightness.clear_with_capacity();
+
+        for (int i = 0; i < 256; i++) {
+            m_data.red.append(0);
+            m_data.green.append(0);
+            m_data.blue.append(0);
+            m_data.brightness.append(0);
+        }
+
+        Color pixel_color;
+        for (int x = 0; x < full_bitmap->width(); x++) {
+            for (int y = 0; y < full_bitmap->height(); y++) {
+                pixel_color = full_bitmap->get_pixel(x, y);
+                if (!pixel_color.alpha()) {
+                    continue;
+                }
+
+                m_data.red[pixel_color.red()]++;
+                m_data.green[pixel_color.green()]++;
+                m_data.blue[pixel_color.blue()]++;
+                m_data.brightness[pixel_color.luminosity()]++;
+            }
+        }
+        int max_brightness_frequency = 0;
+        int max_color_frequency = 0;
+        for (int i = 0; i < 256; i++) {
+            if (m_data.red[i] > max_color_frequency) {
+                max_color_frequency = m_data.red[i];
+            }
```

Comment:
We usually omit curly braces for single-line bodies, like this:

```suggestion
            if (m_data.red[i] > max_color_frequency)
                max_color_frequency = m_data.red[i];
```

same below.